lib/awful/layout/suit/floating.lua

```diff
@@ -33,71 +33,80 @@ function floating.mouse_resize_handler(c, corner, x, y)
-                                          ng = { width = _mouse.x - g.x,
-                                                 height = _mouse.y - g.y }
-                                      elseif corner == "bottom_left" then
-                                          ng = { x = _mouse.x,
-                                                 width = (g.x + g.width) - _mouse.x,
-                                                 height = _mouse.y - g.y }
-                                      elseif corner == "top_left" then
-                                          ng = { x = _mouse.x,
-                                                 width = (g.x + g.width) - _mouse.x,
-                                                 y = _mouse.y,
-                                                 height = (g.y + g.height) - _mouse.y }
-                                      else
-                                          ng = { width = _mouse.x - g.x,
-                                                 y = _mouse.y,
-                                                 height = (g.y + g.height) - _mouse.y }
-                                      end
-                                      if ng.width <= 0 then ng.width = nil end
-                                      if ng.height <= 0 then ng.height = nil end
-                                      if fixed_x then ng.width = g.width ng.x = g.x end
-                                      if fixed_y then ng.height = g.height ng.y = g.y end
-                                      c:geometry(ng)
-                                      -- Get real geometry that has been applied
-                                      -- in case we honor size hints
-                                      -- XXX: This should be rewritten when size
-                                      -- hints are available from Lua.
-                                      local rg = c:geometry()
-
-                                      if corner == "bottom_right" then
-                                          ng = {}
-                                      elseif corner == "bottom_left" then
-                                          ng = { x = (g.x + g.width) - rg.width  }
-                                      elseif corner == "top_left" then
-                                          ng = { x = (g.x + g.width) - rg.width,
-                                                 y = (g.y + g.height) - rg.height }
-                                      else
-                                          ng = { y = (g.y + g.height) - rg.height }
-                                      end
-                                      c:geometry({ x = ng.x, y = ng.y })
-                                      return true
-                                  end
-                              end
-                              return prev_coords.x == _mouse.x and prev_coords.y == _mouse.y
-                          end, corner .. "_corner")
+    capi.mousegrabber.run(function (position)
```

Comment:
Since this table also contains the `.buttons` field, maybe something like `state` or `mouse_state` would make more sense.

Reply:
done